### DISCUSSION TITLE
OpenER with Allen Bradley, incorrect size

### DISCUSSION THREAD

**Discussion by alexbussiere** [EIPStackGroup/OpENer#471] [General]
I have the library running on my device, using an Ethernet IP scanner we can see all the details and structures of the example as expected (correct size). 

When we go to setup a new EIP device in our Allen Bradley PLC, when we set the Output object to a size of 32, we get an error that the "output size is incorrect". Has anyone experienced this issue ?

**Comment by MartinMelikMerkumians** [comment]
Hi,
You need to change the output object size and the according EDS file entry.
TBH I don't do that usually, but I think that should be the issue. The originator will try to create a connection based on the EDS file entry. If you changed the output object, but not the EDS file, this should result in the experienced error

BR,
Martin

**Reply by alexbussiere** [reply]
So that's the really odd part, all of the sizes are set to 32, in the firmware the array size is 32, in the EDS file it is 32, if we manually setup an Ethernet/IP object we set it up as 32. But with all of these we get an error that the size is incorrect in the Allen Bradley PLC. We have been using "EnIPExplorer" to test the device, and the data structures appear as expected there and the array data can be changed and the data shows up as expected there.

**Comment by alexbussiere** [comment]
Running the sample application, I am getting the following message when I request the instance number 154 : 

```
E (1150448) OPENER: NotifyMessageRouter: routing unconnected message

E (1150454) OPENER: NotifyMessageRouter: sending CIP_ERROR_OBJECT_DOES_NOT_EXIST reply, class id 0x9a is not registered
```

but when I request instance 246, the device will reply with expected information and does not generate the same message

**Reply by MartinMelikMerkumians** [reply]
Based on the error message, I would assume you are not specifying the correct class id
